torchdata/datapipes/iter/load/online.py

```diff
@@ -49,30 +49,36 @@ def __len__(self) -> int:
         return len(self.source_datapipe)
 
 
-def _get_response_from_google_drive(url: str) -> Tuple[str, StreamWrapper]:
+def _get_response_from_google_drive(url: str, *, timeout: Optional[float]) -> Tuple[str, StreamWrapper]:
     confirm_token = None
-    session = requests.Session()
-    response = session.get(url, stream=True)
-    for k, v in response.cookies.items():
-        if k.startswith("download_warning"):
-            confirm_token = v
-    if confirm_token is None:
-        if "Quota exceeded" in str(response.content):
-            raise RuntimeError(
-                "Google drive link {} is currently unavailable, because the quota was exceeded.".format(url)
-            )
-
-    if confirm_token:
-        url = url + "&confirm=" + confirm_token
-
-    response = session.get(url, stream=True)
-
-    if "content-disposition" not in response.headers:
-        raise RuntimeError("Internal error: headers don't contain content-disposition.")
-
-    filename = re.findall('filename="(.+)"', response.headers["content-disposition"])
-    if filename is None:
-        raise RuntimeError("Filename could not be autodetected")
+    with requests.Session() as session:
```

Comment:
Quick question here. As we do streaming session here, could using `with` context manager would destroy the response before we actually read data.

Let's add the label `ciflow/slow` to test them.

Reply:
I think you are right, I would think the `with` statement will close the session before the `response.raw` byte stream is fully read. I will work on testing it out.

Reply:
@ejguan The test seems fine. Do you think we should still remove the `with` statements? I believe `response.raw` should be a byte stream but I'm unsure if still has a network connection and if it does, it will be terminated by the closing of `session` (I think it should).

Reply:
I made a mistake, the corresponding label is `ciflow/period` as https://github.com/pytorch/data/blob/7899c86a12613748d14a4fbb352edfb3409afa46/test/test_period.py#L13 is in `test_period.py` file.

Besides, could we add `reset` tests for gdrive and other online datapipe?

Reply:
I think the existing reset test should be testing the behavior?

First, the functional test builds an iterator and reads the first line from `GDriveReader` to confirm it has content.
https://github.com/pytorch/data/blob/7899c86a12613748d14a4fbb352edfb3409afa46/test/test_period.py#L20-L25

After that, we call `HashChecker` on `GDriveReader` and check the full hash of the downloaded file.

https://github.com/pytorch/data/blob/7899c86a12613748d14a4fbb352edfb3409afa46/test/test_period.py#L27-L32

As part of `HashChecker` I believe this line here should reset the `source_datapipe` (equivalent to creating a new iterator) so the content will be read from the beginning again?

https://github.com/pytorch/data/blob/7899c86a12613748d14a4fbb352edfb3409afa46/torchdata/datapipes/iter/util/hashchecker.py#L49


Therefore, the behavior is that one line from `GDriveReader` was read, it gets reset and the full hash is checked. Does that make sense or am I still missing something?

I also just tried running this and the test still passes:
```python
check_cache_dp = gdrive_reader_dp.check_hash({expected_file_name: expected_MD5_hash}, "md5", rewind=False)
it = iter(check_cache_dp)
path, stream = next(it)
self.assertEqual(expected_file_name, os.path.basename(path))
self.assertTrue(io.BufferedReader, type(stream))
it = iter(check_cache_dp)
path, stream = next(it)
self.assertEqual(expected_file_name, os.path.basename(path))
self.assertTrue(io.BufferedReader, type(stream))
```

Reply:
Thank! Interesting use case. I will approve the PR as it's still working with the context manager